sqlmesh_dbt/operations.py

```diff
@@ -109,25 +89,20 @@ def create(
         return DbtOperations(sqlmesh_context, dbt_project)
 
 
-def _inject_default_start_date(project_dir: t.Optional[Path] = None) -> None:
+def init_project_if_required(project_dir: Path) -> None:
     """
-    SQLMesh needs a start date to as the starting point for calculating intervals on incremental models
+    SQLMesh needs a start date to as the starting point for calculating intervals on incremental models, amongst other things
 
     Rather than forcing the user to update their config manually or having a default that is not saved between runs,
-    we can inject it automatically to the dbt_project.yml file
+    we can generate a basic SQLMesh config if it doesnt exist.
+
+    This is preferable to trying to inject config into `dbt_project.yml` because it means we have full control over the file
+    and dont need to worry about accidentally reformatting it or accidentally clobbering other config
     """
-    from sqlmesh.dbt.project import PROJECT_FILENAME, load_yaml
-    from sqlmesh.utils.yaml import dump
-    from sqlmesh.utils.date import yesterday_ds
-
-    project_yaml_path = (project_dir or Path.cwd()) / PROJECT_FILENAME
-    if project_yaml_path.exists():
-        loaded_project_file = load_yaml(project_yaml_path)
-        start_date_keys = ("start", "+start")
-        if "models" in loaded_project_file and all(
-            k not in loaded_project_file["models"] for k in start_date_keys
-        ):
-            loaded_project_file["models"]["+start"] = yesterday_ds()
-            # todo: this may format the file differently, is that acceptable?
-            with project_yaml_path.open("w") as f:
-                dump(loaded_project_file, f)
+    from sqlmesh.cli.project_init import init_example_project, ProjectTemplate
```

Comment:
Shouldn't all these imports be under the check of the file existence? Otherwise doesn't this defeat the purpose?